### PR TITLE
Drop all repeated CLI info commands

### LINKED ISSUE #870: Improve CLI automation to not repeat info command
There are some tests that are running info command two and even three consecutive times.

The factories already call info to retrieve the created object ID and raise an exception if it does not find it. So a consecutive call to info is not required.

Examples:

```
2014-06-09 08:07:15 - robottelo - DEBUG - >>> LANG=en_US hammer -v -u admin -p changeme  --output csv repository create --product-id='18' --name='QvaYbbRBcqxTaGG' --publish-via-http='true' --url='http://someserver.com/fakepuppet01/' --content-type='puppet'
2014-06-09 08:07:19 - robottelo - DEBUG - <<<
Message,Id,Name
Repository created,37,QvaYbbRBcqxTaGG
2014-06-09 08:07:19 - robottelo - DEBUG - >>> LANG=en_US hammer -v -u admin -p changeme  repository info --id='37'
2014-06-09 08:07:22 - robottelo - DEBUG - <<<
ID:                 37
Name:               QvaYbbRBcqxTaGG
Label:              QvaYbbRBcqxTaGG
Organization:       atqeji
Red Hat Repository: no
Content Type:       puppet
URL:                http://someserver.com/fakepuppet01/
Publish Via HTTP:   yes
Published At:       http://server.com/pulp/repos/atqeji/Library/custom/bpjSVzQZSgDtmbxkIGIn/QvaYbbRBcqxTaGG
Product:            
    ID:   18
    Name: sNHFUJdhTWrYmIIWTPyo
GPG Key:            

Sync:               
    Status: Not Synced
Created:            2014/06/09 12:07:16
Updated:            2014/06/09 12:07:17
Content Counts:

2014-06-09 08:07:27 - robottelo - DEBUG - >>> LANG=en_US hammer -v -u admin -p changeme  repository info --id='37'
2014-06-09 08:07:30 - robottelo - DEBUG - <<<
ID:                 37
Name:               QvaYbbRBcqxTaGG
Label:              QvaYbbRBcqxTaGG
Organization:       atqeji
Red Hat Repository: no
Content Type:       puppet
URL:                http://someserver.com/fakepuppet01/
Publish Via HTTP:   yes
Published At:       http://server.com/pulp/repos/atqeji/Library/custom/bpjSVzQZSgDtmbxkIGIn/QvaYbbRBcqxTaGG
Product:            
    ID:   18
    Name: sNHFUJdhTWrYmIIWTPyo
GPG Key:            

Sync:               
    Status: Not Synced
Created:            2014/06/09 12:07:16
Updated:            2014/06/09 12:07:17
Content Counts:

.
```

```
2014-06-09 08:07:30 - robottelo - DEBUG - >>> LANG=en_US hammer -v -u admin -p changeme  --output csv gpg create --organization-id='91' --name='ckpn7viz' --key='/tmp/zs43y99p'
2014-06-09 08:07:32 - robottelo - DEBUG - <<<
Message,Id,Name
GPG Key created,10,ckpn7viz
2014-06-09 08:07:32 - robottelo - DEBUG - >>> LANG=en_US hammer -v -u admin -p changeme  --output csv gpg info --id='10' --organization-id='91'
2014-06-09 08:07:35 - robottelo - DEBUG - <<<
ID,Name,Organization,Content
10,ckpn7viz,atqeji,yrdymdplx2whswgoxarjcwlpux32vaep5nsv
2014-06-09 08:07:41 - robottelo - DEBUG - >>> LANG=en_US hammer -v -u admin -p changeme  --output csv repository create --product-id='18' --name='<sub>ilFEeLHOzreEQSo</sub>' --gpg-key-id='10' --publish-via-http='true' --url='http://someserver.com/fakerepo01/' --content-type='yum'
2014-06-09 08:07:46 - robottelo - DEBUG - <<<
Message,Id,Name
Repository created,38,<sub>ilFEeLHOzreEQSo</sub>
2014-06-09 08:07:46 - robottelo - DEBUG - >>> LANG=en_US hammer -v -u admin -p changeme  repository info --id='38'
2014-06-09 08:07:49 - robottelo - DEBUG - <<<
ID:                 38
Name:               <sub>ilFEeLHOzreEQSo</sub>
Label:              _sub_ilFEeLHOzreEQSo__sub_
Organization:       atqeji
Red Hat Repository: no
Content Type:       yum
URL:                http://someserver.com/fakerepo01/
Publish Via HTTP:   yes
Published At:       http://server.com/pulp/repos/atqeji/Library/custom/bpjSVzQZSgDtmbxkIGIn/_sub_ilFEeLHOzreEQSo__sub_
Product:            
    ID:   18
    Name: sNHFUJdhTWrYmIIWTPyo
GPG Key:            
    ID:   10
    Name: ckpn7viz
Sync:               
    Status: Not Synced
Created:            2014/06/09 12:07:42
Updated:            2014/06/09 12:07:44
Content Counts:

2014-06-09 08:07:54 - robottelo - DEBUG - >>> LANG=en_US hammer -v -u admin -p changeme  repository info --id='38'
2014-06-09 08:07:57 - robottelo - DEBUG - <<<
ID:                 38
Name:               <sub>ilFEeLHOzreEQSo</sub>
Label:              _sub_ilFEeLHOzreEQSo__sub_
Organization:       atqeji
Red Hat Repository: no
Content Type:       yum
URL:                http://someserver.com/fakerepo01/
Publish Via HTTP:   yes
Published At:       http://server.com/pulp/repos/atqeji/Library/custom/bpjSVzQZSgDtmbxkIGIn/_sub_ilFEeLHOzreEQSo__sub_
Product:            
    ID:   18
    Name: sNHFUJdhTWrYmIIWTPyo
GPG Key:            
    ID:   10
    Name: ckpn7viz
Sync:               
    Status: Not Synced
Created:            2014/06/09 12:07:42
Updated:            2014/06/09 12:07:44
Content Counts:

2014-06-09 08:07:57 - robottelo - DEBUG - >>> LANG=en_US hammer -v -u admin -p changeme  repository info --id='38'
2014-06-09 08:07:59 - robottelo - DEBUG - <<<
ID:                 38
Name:               <sub>ilFEeLHOzreEQSo</sub>
Label:              _sub_ilFEeLHOzreEQSo__sub_
Organization:       atqeji
Red Hat Repository: no
Content Type:       yum
URL:                http://someserver.com/fakerepo01/
Publish Via HTTP:   yes
Published At:       http://server.com/pulp/repos/atqeji/Library/custom/bpjSVzQZSgDtmbxkIGIn/_sub_ilFEeLHOzreEQSo__sub_
Product:            
    ID:   18
    Name: sNHFUJdhTWrYmIIWTPyo
GPG Key:            
    ID:   10
    Name: ckpn7viz
Sync:               
    Status: Not Synced
Created:            2014/06/09 12:07:42
Updated:            2014/06/09 12:07:44
Content Counts:

.
```

### PR DESCRIPTION
**Pull request by elyezer**
CLI create already run info command after creating the entity, because
this is not necessary to run info again.

Closes #870

### CODE REVIEW

File: tests/foreman/cli/test_contentviews.py [hunk 1]
```diff
@@ -187,19 +176,14 @@ def test_cv_create_cli_negative(self, test_data):
...
-        org_obj = make_org(cached=True)
-
-        result = Org.info({'id': org_obj['id']})
-        self.assertEqual(result.return_code, 0, "Failed to create object")
-        self.assertEqual(
-            len(result.stderr), 0, "There should not be an exception here")
-
-        test_data['organization-id'] = org_obj['id']
         result = ContentView.create(test_data)
         self.assertNotEqual(result.return_code, 0)
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: Should this be removed?

File: tests/foreman/cli/test_contentviews.py [hunk 2]
```diff
@@ -692,39 +617,35 @@ def test_cv_associate_components_composite_negative(self):
...
+        })
+        self.assertEqual(result.return_code, 0)
+        self.assertEqual(len(result.stderr), 0)
 
         # Publish a new version of CV
         result = ContentView.publish({u'id': new_cv['id']})
-        self.assertEqual(result.return_code, 0,
-                         "Publish CV was not successful")
+        self.assertEqual(
+            result.return_code, 0, "Publish CV was not successful")
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: You have removed the third parameter, the error message that gets displayed to the user, in other parts of the code, so perhaps you should remove this one here too?

File: tests/foreman/cli/test_contentviews.py [hunk 3]
```diff
@@ -692,39 +617,35 @@ def test_cv_associate_components_composite_negative(self):
...
-        cv_version = ContentView.version_list(
-            {
-                u'content-view-id': new_cv['id']
-            }
+        cv_version = ContentView.version_list({
+            u'content-view-id': new_cv['id']
+        })
+        self.assertEqual(
+            cv_version.return_code, 0,
+            "Version list for selected CV was not found"
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: You have removed the third parameter, the error message that gets displayed to the user, in other parts of the code, so perhaps you should remove this one here too?

File: tests/foreman/cli/test_contentviews.py [hunk 4]
```diff
@@ -983,16 +902,22 @@ def test_cv_promote_custom_content(self):
...
-                                             u'repository-id': new_repo['id']})
-        self.assertEqual(result.return_code, 0,
-                         "Repo was not associated to selected CV")
+        result = ContentView.add_repository({
+            u'id': new_cv['id'],
+            u'repository-id': new_repo['id'],
+        })
+        self.assertEqual(
+            result.return_code, 0,
+            "Repo was not associated to selected CV"
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: You have removed the third parameter, the error message that gets displayed to the user, in other parts of the code, so perhaps you should remove this one here too?

File: tests/foreman/cli/test_contentviews.py [hunk 5]
```diff
@@ -983,16 +902,22 @@ def test_cv_promote_custom_content(self):
...
+        )
         self.assertEqual(len(result.stderr), 0, "No error was expected")
 
         # Publish a new version of CV
         result = ContentView.publish({u'id': new_cv['id']})
-        self.assertEqual(result.return_code, 0,
-                         "Publishing a new version of CV was not successful")
+        self.assertEqual(
+            result.return_code, 0,
+            "Publishing a new version of CV was not successful"
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: You have removed the third parameter, the error message that gets displayed to the user, in other parts of the code, so perhaps you should remove this one here too?

File: tests/foreman/cli/test_contentviews.py [hunk 6]
```diff
@@ -1004,8 +929,10 @@ def test_cv_promote_custom_content(self):
             u'id': result.stdout['versions'][0]['id'],
             u'to-lifecycle-environment-id': self.env1['id'],
         })
-        self.assertEqual(result.return_code, 0,
-                         "Promoting a version of CV was not successful")
+        self.assertEqual(
+            result.return_code, 0,
+            "Promoting a version of CV was not successful"
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: You have removed the third parameter, the error message that gets displayed to the user, in other parts of the code, so perhaps you should remove this one here too?

File: tests/foreman/cli/test_contentviews.py [hunk 7]
```diff
@@ -1055,16 +980,22 @@ def test_cv_promote_composite(self):
...
-                                             u'repository-id': new_repo['id']})
-        self.assertEqual(result.return_code, 0,
-                         "Repo was not associated to selected CV")
+        result = ContentView.add_repository({
+            u'id': new_cv['id'],
+            u'repository-id': new_repo['id'],
+        })
+        self.assertEqual(
+            result.return_code, 0,
+            "Repo was not associated to selected CV"
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: You have removed the third parameter, the error message that gets displayed to the user, in other parts of the code, so perhaps you should remove this one here too?

File: tests/foreman/cli/test_contentviews.py [hunk 8]
```diff
@@ -1055,16 +980,22 @@ def test_cv_promote_composite(self):
...
+        )
         self.assertEqual(len(result.stderr), 0, "No error was expected")
 
         # Publish a new version of CV
         result = ContentView.publish({u'id': new_cv['id']})
-        self.assertEqual(result.return_code, 0,
-                         "Publishing a new version of CV was not successful")
+        self.assertEqual(
+            result.return_code, 0,
+            "Publishing a new version of CV was not successful"
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: You have removed the third parameter, the error message that gets displayed to the user, in other parts of the code, so perhaps you should remove this one here too?

File: tests/foreman/cli/test_contentviews.py [hunk 9]
```diff
@@ -1088,14 +1019,18 @@ def test_cv_promote_composite(self):
             u'id': con_view['id'],
             u'content-view-version-id': version1_id,
         })
-        self.assertEqual(result.return_code, 0,
-                         "Repo was not associated to selected CV")
+        self.assertEqual(
+            result.return_code, 0,
+            "Repo was not associated to selected CV"
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: You have removed the third parameter, the error message that gets displayed to the user, in other parts of the code, so perhaps you should remove this one here too?

File: tests/foreman/cli/test_lifecycleenvironment.py [hunk 10]
```diff
@@ -62,37 +62,24 @@ def test_bugzilla_1077333(self):
...
-        payload = {
+        data = {
             'organization-id': self.org['id'],
             'name': gen_string('utf8', 15),
         }
 
-        new_obj = make_lifecycle_environment(payload)
-        self.assertIsNotNone(
-            new_obj, "Could not create lifecycle environment.")
+        result = LifecycleEnvironment.create(data)
```
Comment (omaciel) [inline tests/foreman/cli/test_lifecycleenvironment.py]: Why not use the factory method?

File: tests/foreman/cli/test_org.py [hunk 11]
```diff
@@ -226,36 +207,17 @@ def test_bugzilla_1076568_2(self):
         @BZ: 1076568
 
         """
+        org = make_org()
 
-        new_obj = make_org()
-
-        # Can we find the new object?
```
Comment (omaciel) [inline tests/foreman/cli/test_org.py]: Isn't this a valid test? You just deleted the entity and now you make sure that you cannot find it?

File: tests/foreman/cli/test_org.py [hunk 12]
```diff
@@ -420,23 +351,17 @@ def test_bugzilla_1023125(self):
...
-        result = Org.info({'id': new_obj['id']})
-        self.assertEqual(result.return_code, 0, "Failed to create object")
-        self.assertEqual(len(result.stderr), 0,
-                         "There should not be an exception here")
-        self.assertEqual(new_obj['name'], result.stdout['name'])
+        org = make_org()
 
         # Create new org with the same name as before
         # should yield an exception
         with self.assertRaises(Exception):
```
Comment (omaciel) [inline tests/foreman/cli/test_org.py]: ``self.assertRaises(CLIFactoryError)``?

**Comment by omaciel** [comment]
We need to determine what is the best practice to handle assertions and providing information to the user. This PR shows that some assertions only use 2 parameters for comparison (i.e. ``assertEqual``, ``assertGreater``, etc) but we also have several places where a third parameter is passed, a user friendly message to be displayed to the user. We need to determine how we want to write our assertions and stick to it, though I will not block this PR from being merged.

Also, ``test_redmine_4271`` was removed but there was no mention as to the why... can you add a comment?

File: tests/foreman/cli/test_gpgkey.py [hunk 13]
```diff
@@ -79,41 +79,6 @@ def create_gpg_key_file(self, content=None):
 
     # Bug verification
 
-    @skip_if_bug_open('redmine', 4271)
-    def test_redmine_4271(self):
```
Comment (elyezer) [inline tests/foreman/cli/test_gpgkey.py]: There is no special check for the CSV output in this test. Also CSV output is being used every time a GPG key is created because the default CLI behaviour. With that said there is no need to this test exist because it is doing a positive create test which already exists.

File: tests/foreman/cli/test_org.py (hunk 12 above)
Comment (elyezer) [inline tests/foreman/cli/test_org.py]: This is out of scope of this PR, though I have changed the code parts I've touched. I have created #2055 to handle this.

**Comment by elyezer** [comment]
The message is only useful if it is meaningful. Some cases where it is not: return_code != 0 already says that there is an error, length stderror > 0 already says that there is an error. But `Repo was not associated to selected CV` have a meaning and let you know that the content view association failed. This could be a good practice if used wisely.

File: tests/foreman/cli/test_contentviews.py (hunk 2 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: https://github.com/SatelliteQE/robottelo/pull/2054#issuecomment-92175686

File: tests/foreman/cli/test_contentviews.py (hunk 3 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: https://github.com/SatelliteQE/robottelo/pull/2054#issuecomment-92175686

File: tests/foreman/cli/test_contentviews.py (hunk 5 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: https://github.com/SatelliteQE/robottelo/pull/2054#issuecomment-92175686

File: tests/foreman/cli/test_contentviews.py (hunk 6 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: https://github.com/SatelliteQE/robottelo/pull/2054#issuecomment-92175686

File: tests/foreman/cli/test_contentviews.py (hunk 7 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: https://github.com/SatelliteQE/robottelo/pull/2054#issuecomment-92175686

File: tests/foreman/cli/test_lifecycleenvironment.py (hunk 10 above)
Comment (elyezer) [inline tests/foreman/cli/test_lifecycleenvironment.py]: Will revert to the factory. When I saw this I miss thought that create just created the entity and not run the info, and this is not true. Thanks for catching this.

File: tests/foreman/cli/test_contentviews.py (hunk 1 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Nice catch. Actually this should be updated to:

```python
with self.assertRaises(CLIFactoryError):
    make_content_view(test_data)
```

After updating here is the test result with smoke flag equals 0:

```
$ py.test tests/foreman/cli/test_contentviews.py -ktest_cv_create_cli_negative
================================== test session starts ==================================
platform darwin -- Python 2.7.6 -- py-1.4.26 -- pytest-2.6.4
collected 52 items

tests/foreman/cli/test_contentviews.py .......

================ 45 tests deselected by '-ktest_cv_create_cli_negative' =================
======================= 7 passed, 45 deselected in 57.37 seconds ========================
```

File: tests/foreman/cli/test_contentviews.py [hunk 14]
```diff
@@ -157,23 +156,13 @@ def test_cv_create_cli(self, test_data):
...
-        result = Org.info({'id': org_obj['id']})
-        self.assertEqual(result.return_code, 0, "Failed to create object")
-        self.assertEqual(
-            len(result.stderr), 0, "There should not be an exception here")
-
-        test_data['organization-id'] = org_obj['id']
-
         try:
-            con_view = make_content_view(test_data)
+            test_data['organization-id'] = make_org(cached=True)
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_contentviews.py]: Should this be `make_org(cached=True)['id']`?

File: tests/foreman/cli/test_contentviews.py [hunk 15]
```diff
@@ -692,39 +616,32 @@ def test_cv_associate_components_composite_negative(self):
...
+        self.assertEqual(result.return_code, 0)
+        self.assertEqual(len(result.stderr), 0)
 
         # Publish a new version of CV
         result = ContentView.publish({u'id': new_cv['id']})
-        self.assertEqual(result.return_code, 0,
-                         "Publish CV was not successful")
-        self.assertEqual(len(result.stderr), 0, "No error was expected")
+        self.assertEqual(
+            result.return_code, 0)
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_contentviews.py]: Nitpick: place this statement on one line.

File: tests/foreman/cli/test_contentviews.py [hunk 16]
```diff
@@ -1088,34 +1000,33 @@ def test_cv_promote_composite(self):
             u'id': con_view['id'],
             u'content-view-version-id': version1_id,
         })
-        self.assertEqual(result.return_code, 0,
-                         "Repo was not associated to selected CV")
-        self.assertEqual(len(result.stderr), 0, "No error was expected")
+        self.assertEqual(
+            result.return_code, 0,
+            "Repo was not associated to selected CV"
+        )
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_contentviews.py]: Are we keeping this error message because it's especially useful?

File: tests/foreman/cli/test_contentviews.py [hunk 17]
```diff
@@ -1482,22 +1388,25 @@ def test_cv_publish_version_changes_in_target_env(self):
...
+        self.assertEqual(len(result.stderr), 0)
 
         # Publish a version1 of CV
         result = ContentView.publish({u'id': new_cv['id']})
-        self.assertEqual(result.return_code, 0,
-                         "Publishing version1 of CV was not successful")
-        self.assertEqual(len(result.stderr), 0, "No error was expected")
+        self.assertEqual(
+            result.return_code, 0,
+            "Publishing version1 of CV was not successful"
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_contentviews.py]: Is this message being kept because it's especially useful?

File: tests/foreman/cli/test_contentviews.py (hunk 16 above)
Comment (Ichimonji10) [inline tests/foreman/cli/test_contentviews.py]: The same question applies to several locations throughout this PR.

File: tests/foreman/cli/test_gpgkey.py [hunk 18]
```diff
@@ -125,10 +90,11 @@ def test_redmine_4272(self):
         @BZ: Redmine#4272
 
         """
-
         # GPG Key data
-        data = {'name': gen_string("alpha", 10)}
-        data['organization-id'] = self.org['id']
+        data = {
+            'name': gen_string("alpha", 10),
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_gpgkey.py]: `s/"/'/g`
`s/, 10//`?

File: tests/foreman/cli/test_template.py [hunk 19]
```diff
@@ -26,24 +26,17 @@ def test_create_template_1(self):
...
         try:
-            new_obj = make_template(
-                {
-                    'name': name,
-                    'content': content,
-                }
-            )
+            new_obj = make_template({
+                'name': name,
+                'content': gen_string("alpha", 10),
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_template.py]: `s/"/'/g`
`s/, 10//`? (Signature of [`gen_alpha`](https://github.com/omaciel/fauxfactory/blob/master/fauxfactory/__init__.py#L147))

File: tests/foreman/cli/test_template.py [hunk 20]
```diff
@@ -53,27 +46,21 @@ def test_update_template_1(self):
...
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_template.py]: `s/"/'/g`
`s/, 10//`? (Signature of [`gen_alpha`](https://github.com/omaciel/fauxfactory/blob/master/fauxfactory/__init__.py#L147))

File: tests/foreman/cli/test_template.py [hunk 21]
```diff
@@ -53,27 +46,21 @@ def test_update_template_1(self):
...
+            })
         except CLIFactoryError as e:
             self.fail(e)
 
-        result = Template.info({'id': new_obj['id']})
+        updated_name = gen_string("alpha", 10)
```
Comment (Ichimonji10) [inline tests/foreman/cli/test_template.py]: `s/"/'/g`
`s/, 10//`? (Signature of [`gen_alpha`](https://github.com/omaciel/fauxfactory/blob/master/fauxfactory/__init__.py#L147))

**Comment by Ichimonji10** [comment]
ACK pending comments.

I like how much more concise the code is.

File: tests/foreman/cli/test_contentviews.py (hunk 14 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Good catch, will update and run the test

File: tests/foreman/cli/test_contentviews.py (hunk 15 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Missed that, thank you.

File: tests/foreman/cli/test_contentviews.py (hunk 16 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: I have reviewed some of the messages, will do it again. There are many files and tests to look over.

File: tests/foreman/cli/test_gpgkey.py (hunk 18 above)
Comment (elyezer) [inline tests/foreman/cli/test_gpgkey.py]: Not touched this to maintain focus on the PR purpose, but I think worth changing this and will handle.

File: tests/foreman/cli/test_contentviews.py [hunk 22]
```diff
@@ -154,23 +153,13 @@ def test_cv_create_cli(self, test_data):
...
-        self.assertEqual(result.return_code, 0, "Failed to create object")
-        self.assertEqual(
-            len(result.stderr), 0, "There should not be an exception here")
-
-        test_data['organization-id'] = org_obj['id']
-
         try:
-            con_view = make_content_view(test_data)
+            test_data['organization-id'] = make_org(cached=True)['id']
+            content_view = make_content_view(test_data)
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: The method ``make_content_view`` takes a dictionary but we're passing only an ID?

File: tests/foreman/cli/test_contentviews.py [hunk 23]
```diff
@@ -184,19 +173,13 @@ def test_cv_create_cli_negative(self, test_data):
...
-        self.assertEqual(
-            len(result.stderr), 0, "There should not be an exception here")
-
-        test_data['organization-id'] = org_obj['id']
-        result = ContentView.create(test_data)
-        self.assertNotEqual(result.return_code, 0)
-        self.assertGreater(len(result.stderr), 0,
-                           "There should be an exception here")
+        with self.assertRaises(CLIFactoryError):
+            make_content_view(test_data)
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: The method ``make_content_view`` takes a dictionary but we're passing only an ID?

File: tests/foreman/cli/test_contentviews.py [hunk 24]
```diff
@@ -1479,22 +1366,25 @@ def test_cv_publish_version_changes_in_target_env(self):
...
+        self.assertEqual(len(result.stderr), 0)
 
         # Publish a version1 of CV
         result = ContentView.publish({u'id': new_cv['id']})
-        self.assertEqual(result.return_code, 0,
-                         "Publishing version1 of CV was not successful")
-        self.assertEqual(len(result.stderr), 0, "No error was expected")
+        self.assertEqual(
+            result.return_code, 0,
+            "Publishing version1 of CV was not successful"
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: Was this message left on purpose?

File: tests/foreman/cli/test_contentviews.py [hunk 25]
```diff
@@ -1502,14 +1392,15 @@ def test_cv_publish_version_changes_in_target_env(self):
         # Actual assert for this test happens HERE
         # Test whether the version1 now belongs to Library
         result_version = ContentView.version_info({u'id': version1_id})
-        self.assertEqual(result_version.return_code, 0,
-                         "ContentView version was not found")
-        self.assertEqual(len(result_version.stderr), 0,
-                         "No error was expected")
+        self.assertEqual(
+            result_version.return_code, 0,
+            "ContentView version 1 was not found"
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: Was this message left on purpose?

File: tests/foreman/cli/test_contentviews.py [hunk 26]
```diff
@@ -1502,14 +1392,15 @@ def test_cv_publish_version_changes_in_target_env(self):
...
+        )
+        self.assertEqual(len(result_version.stderr), 0)
         self.assertEqual(
             result_version.stdout['lifecycle-environments'][0]['label'],
             u'Library',
-            'This version not in Library'
+            'Version 1 is not in Library'
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: Was this message left on purpose?

File: tests/foreman/cli/test_contentviews.py [hunk 27]
```diff
@@ -1832,19 +1749,14 @@ def test_cv_roles_admin_user_negative(self, test_data):
...
-        org_obj = make_org(cached=True)
-
-        result = Org.info({'id': org_obj['id']})
-        self.assertEqual(result.return_code, 0, "Failed to create object")
-        self.assertEqual(
-            len(result.stderr), 0, "There should not be an exception here")
-
-        test_data['organization-id'] = org_obj['id']
+        try:
+            test_data['organization-id'] = make_org(cached=True)['id']
```
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: Do we only want the ``id`` here? Don't we need a dictionary for the next time this is used?

File: tests/foreman/cli/test_host_collection.py [hunk 28]
```diff
@@ -455,35 +348,27 @@ def test_add_content_host(self):
...
-            len(result.stderr), 0, 'There should not be an error here')
-        self.assertGreater(result.stdout['total-content-hosts'],
-                           no_of_content_host,
-                           "There should not be an exception here")
+        self.assertEqual(result.return_code, 0)
+        self.assertEqual(len(result.stderr), 0)
+        self.assertGreater(
+            result.stdout['total-content-hosts'],
+            no_of_content_host,
+            "There should not be an exception here"
```
Comment (omaciel) [inline tests/foreman/cli/test_host_collection.py]: This message is kind of meh...

File: tests/foreman/cli/test_lifecycleenvironment.py [hunk 29]
```diff
@@ -286,55 +221,25 @@ def test_positive_delete_1(self, test_data):
...
         self.assertEqual(
-            len(result.stderr), 0, "There should not be an error here.")
-
-        # Can we find the object
-        result = LifecycleEnvironment.info(
-            {
-                'organization-id': self.org['id'],
-                'id': new_obj['id'],
-            }
+            len(result.stderr), 0, "There should not be an error here."
```
Comment (omaciel) [inline tests/foreman/cli/test_lifecycleenvironment.py]: This message could be improved (I know this is not part of the PR per se)

**Comment by omaciel** [comment]
I reviewed it again and I noticed that most messages used in assertions have been removed with a few exceptions. Safe to assume that you felt those messages would be more helpful to anyone reviewing failures?

Out of the remaining messages mentioned above, there are 2 that perhaps could either be improved or altogether removed?

Finally, there are some calls to ``make_content_view`` that instead of passing a dictionary are not passing an ``id``. I think we need to revert it to passing the dictionary.

File: tests/foreman/cli/test_contentviews.py (hunk 24 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Yes, we have two versions being created on this test, and I decided to leave this message to make easier to know if the first or second version that failed.

File: tests/foreman/cli/test_contentviews.py (hunk 25 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Yes, we have two versions being created on this test, and I decided to leave this message to make easier to know if the first or second version that failed.

File: tests/foreman/cli/test_contentviews.py (hunk 26 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Yes, we have two versions being created on this test, and I decided to leave this message to make easier to know if the first or second version that failed.

File: tests/foreman/cli/test_contentviews.py (hunk 27 above)
Comment (elyezer) [inline tests/foreman/cli/test_contentviews.py]: Only the ID is required and it is used once.

File: tests/foreman/cli/test_host_collection.py (hunk 28 above)
Comment (elyezer) [inline tests/foreman/cli/test_host_collection.py]: Thanks!

File: tests/foreman/cli/test_lifecycleenvironment.py (hunk 29 above)
Comment (elyezer) [inline tests/foreman/cli/test_lifecycleenvironment.py]: Will remove it, thanks

File: tests/foreman/cli/test_contentviews.py (hunk 27 above)
Comment (omaciel) [inline tests/foreman/cli/test_contentviews.py]: Doh! Yeah, missed that, sorry.

**Comment by omaciel** [comment]
APT

**Comment by elyezer** [comment]
Edit: rebased
